src/HealthChecks.Kafka/DependencyInjection/KafkaHealthCheckBuilderExtensions.cs

```diff
@@ -113,4 +113,36 @@ public static IHealthChecksBuilder AddKafka(
             tags,
             timeout));
     }
+
+    /// <summary>
+    /// Add a health check for Kafka cluster.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="optionFactory">Options to configure Kafka health check.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'kafka' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddKafka(
+        this IHealthChecksBuilder builder,
+       Func<IServiceProvider, KafkaHealthCheckOptions> optionFactory,
+        string? name = default,
+        HealthStatus? failureStatus = default,
+        IEnumerable<string>? tags = default,
+        TimeSpan? timeout = default)
+    {
+        Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNull(optionFactory);
+
+        return builder.Add(new HealthCheckRegistration(
+            name ?? NAME,
+            sp => new KafkaHealthCheck(optionFactory.Invoke(sp)),

```

Comment:
I think this callback gets invoked every time the health is checked. So the optionFactory would be invoked each time as well.

Reply:
Correct.